docs/training/apis-howto/api-pytorch-trainer-ug.rst

```diff
@@ -0,0 +1,354 @@
+          _union_key = "type"
+
+      @MyUnion.member("a")
+      class MemberA(MyUnion):
+          _id = "..."
+
+.. code::
+
+   from determined import pytorch
+
+   def main():
+       # pytorch.init() returns a PyTorchTrialContext for instantiating PyTorchTrial
+       with det.pytorch.init() as train_context:
+           trial = MyPyTorchTrial(train_context)
+           trainer = det.pytorch.Trainer(trial, train_context)
+
+           # (Optional) Configure Determined profiler before calling .fit()
+           trainer.configure_profiler(enabled=True,
+                                      sync_timings=True,
+                                      begin_on_batch=0,
+                                      end_after_batch=10)
+
+           # Train
+           trainer.fit(
+               checkpoint_period=pytorch.Batch(10),
+               validation_period=pytorch.Batch(10),
+           )
+
+   if __name__ == "__main__":
+       # Configure logging here instead of through the expconf
+       logging.basicConfig(level=logging.INFO, format=det.LOG_FORMAT)
+       main()
+
+******************************************
+ Step 3: Run Your Training Script Locally
+******************************************
+
+With the PyTorch Trainer API, you can run training scripts locally without needing to use an
+experiment configuration (YAML) file. Be sure to specify max_length in the .fit() call, and
+global_batch_size in pytorch.init().
+
+Run this script directly (python3 train.py), or inside of a Jupyter notebook.
+
+.. code::
+
+   + hparams = {"global_batch_size": 32, "lr": 0.02}
+   + expconf = yaml.safe_load(pathlib.Path("./det.yaml").read_text())
+
+   # hparams and exp_conf are optional. Only needed by init() if training code calls
+   # context.get_hparams() or context.get_experiment_config()
+   + with det.pytorch.init(hparams=hparams, exp_conf=expconf) as train_context:
+         # (Optional) Preferred way to access hparams in the Trial
+   +     trial = MyPytorchTrial(train_context, hparams)
+         trainer = det.pytorch.Trainer(trial, train_context)
+         trainer.fit(
+   +         max_length=pytorch.Epoch(1),
+             checkpoint_period=pytorch.Batch([2,5]),
+             validation_period=pytorch.Batch(10),
+       )
+
+Local + Distributed Training
+============================
+
+Local training can utilize multiple GPUs on a single node with a few modifications to the above
+code.
+
+.. note::
+
+   Only Horovod and PyTorch Distributed backends are supported.
+
+.. code::
+
+    def main():
+   +     # Initialize distributed backend before pytorch.init()
+   +     dist.init_process_group(backend="gloo|nccl")
+
+   +     # Set flag used by internal PyTorch training loop
+   +     os.environ["USE_TORCH_DISTRIBUTED"] = "true"
+
+   +     # Initialize DistributedContext specifying chief IP
+         with det.pytorch.init(
+   +       distributed=core.DistributedContext.from_torch_distributed  (chief_ip="localhost")
```

Comment:
weird spaces before `  (chief_ip`

Reply:
side note, `chief_ip="localhost"` is useless to specify.  Either you are all one one node and `chief_ip` is not necessary, or you are on multiple nodes and `chief_ip` must not be `"localhost"`